### ISSUE TITLE
Stuck on boot - apfs_module_start

### ISSUE THREAD

**Issue by sergvax** [banhbaoxamlan/X230-Hackintosh#57]
release 0.7 and 0.5 not working on x230t tablet
stuck on boot - before apple logo - with some white text on black screen
I didn't make a Screenshot of text on black screen

**Comment by banhbaoxamlan** [comment]
Sorry, bro, I can't help you without information. :sweat:

**Comment by ghost** [comment]
Tried to boot and got stuck too
![WhatsApp Image 2021-06-11 at 3 03 22 PM](https://user-images.githubusercontent.com/28967978/121730561-3a934f00-cac6-11eb-925c-df60c0bc8df4.jpeg)

**Comment by Benya9669** [comment]
> Tried to boot and got stuck too

Got same screen
![DSC_0011](https://user-images.githubusercontent.com/60983777/121731289-6bd94280-caf9-11eb-8201-442cb08d2bc6.JPG)

**Comment by banhbaoxamlan** [comment]
Open Config.split: `Kernel -> Quirk -> ExternalDiskIcons`, set `<false/>`.

**Comment by Benya9669** [comment]
Same
![DSC_0012](https://user-images.githubusercontent.com/60983777/121734188-1ef76b00-cafd-11eb-9213-33cd0193f4c1.JPG)

**Comment by banhbaoxamlan** [comment]
[EFI.zip](https://github.com/banhbaoxamlan/X230-Hackintosh/files/6640485/EFI.zip)
Try with this EFI, I have disabled all APCI and kexts.
Go to BIOS Settings, Load Default Option and save.

**Comment by banhbaoxamlan** [comment]
I tested with my X230 (2 machines) but not get this stuck 😓

**Comment by sergvax** [comment]
I tried to update opencore manually - last opencore that works was 0.66, after 0.67-0.7 didn’t - always stuck on

“apfs_module_start”

**Comment by banhbaoxamlan** [comment]
> I tried to update opencore manually - last opencore that works was 0.66, after 0.67-0.7 didn’t - always stuck on
> 
> “apfs_module_start”

https://dortania.github.io/OpenCore-Install-Guide/troubleshooting/extended/kernel-issues.html#stuck-on-rtc-pci-configuration-begins-previous-shutdown-hpet-hid-legacy
- IRQ conflict
- SATA issue
- RTC Missing

**Comment by ga3tano** [comment]
Same as Benya9669, x230 here

**Comment by ga3tano** [comment]
I have the same issue on x230, used to have no problems with 0.4; my unit has:

i7 3520m
original centrino wifi card and original bluetooth module
ssd samsung 860 evo
modded ivy rain bios
X220 keyboard with modded EC
Fingerprint reader (I know it’s disabled but maybe it’s something to consider to find the issue)

![IMG_5559](https://user-images.githubusercontent.com/22911629/121790171-e6c75980-cbdc-11eb-93d8-3cfd5d97f1c4.jpeg)

let me know if I can give you more info, and as always thanks for your amazing work :)

**Comment by maxxwall** [comment]
X230 i5-3320M 
BIOS:2.77 Mod
SSD:UNIS S100 120G
Latest EFI file stuck on “apfs_module_start”
![IMG_20210617_104303](https://user-images.githubusercontent.com/24955785/122325783-c93c1c00-cf5d-11eb-99ef-7c3de0fa37a0.jpg)

[EFI.zip](https://ws28.cn/f/5opsb04glns)
Old EFI files boot normally

**Comment by wahidikqbal** [comment]
Are mini display port on Thinkpad X230 work?

**Comment by 13631181823** [comment]
x230i the same issue ,RC5

**Comment by 13631181823** [comment]
using the acpi and kext from RC5, it boot normally from OC0.65.
but will stuck in OC0.7(and older)
the same conclution with 

> I tried to update opencore manually - last opencore that works was 0.66, after 0.67-0.7 didn’t - always stuck on
> 
> “apfs_module_start”

**Comment by Valeria** [comment]
> Are mini display port on Thinkpad X230 work?

@wahidikqbal Yes it does (at least on my machine). Am running old EFI folder (I believe it's version 0.4)

**Comment by Cikappa2904** [comment]
> using the acpi and kext from RC5, it boot normally from OC0.65.
> but will stuck in OC0.7(and older)
> the same conclution with
> 
> > I tried to update opencore manually - last opencore that works was 0.66, after 0.67-0.7 didn’t - always stuck on
> > “apfs_module_start”

I am having the same issue

**Comment by ghost** [comment]
I also have this problem. I have used the BIOS patch

**Comment by tupinikeebs** [comment]
Any fix yet?

**Comment by Jan-E** [comment]
Try the fix for the boot problem that I had, described here: https://github.com/banhbaoxamlan/X230-Hackintosh/issues/67#issuecomment-894389362

Put that ‘T530’ EFI on a USB drive and boot just one time with it. Then try your normal way of booting once again.

**Comment by Cikappa2904** [comment]
> Try the fix for the boot problem that I had, described here: [#67 (comment)](https://github.com/banhbaoxamlan/X230-Hackintosh/issues/67#issuecomment-894389362)
> 
> Put that ‘T530’ EFI on a USB drive and boot just one time with it. Then try your normal way of booting once again.

i don't know how, i don't know why, but it worked.
Boot that EFI, then boot the non functioning EFI and now it works. this doesn't make any sense but it works ...

**Comment by Jan-E** [comment]
As I said in the Monterey issue:

> I really do not know why the workaround works, but apparently it clears something in my system to fix the 'dead IGPU' issue.

**Comment by ghost** [comment]
> Try the fix for the boot problem that I had, described here: [#67 (comment)](https://github.com/banhbaoxamlan/X230-Hackintosh/issues/67#issuecomment-894389362)
> 
> Put that ‘T530’ EFI on a USB drive and boot just one time with it. Then try your normal way of booting once again.

thank you! With this method, the problem has been solved!

**Comment by Jan-E** [comment]
Quote from the changelog in https://github.com/5T33Z0/Lenovo-T530-Hackinosh-OpenCore/releases/tag/072.2

> Changed UEFI > APFS > MinDate and MinVersion to -1 → Enables APFS Driver loading for any macOS Version. Otherwise, if you use macOS lower than Big Sur and the values are left to defaults (set to 0) you won't see your macOS Partition. NOTE: For security reasons you should change these values according to the macOS version you are running.

This might give a clue.

**Comment by mcanevet** [comment]
When using master + OpenCore files from https://github.com/banhbaoxamlan/X230-Hackintosh/commit/9b80582f4e471b2f4e3021cc1255444e2e39965b it works fine.

```
EFI/BOOT/BOOTx64.efi
EFI/OC/Drivers/OpenCanopy.efi
EFI/OC/Drivers/OpenRuntime.efi
EFI/OC/OpenCore.efi
EFI/OC/Tools/OpenShell.efi
```
I don't really know which exact version of OpenCore it points to because, unless I'm wrong, both are daily builds of unreleased 0.7.1.

**Comment by truongvantuan** [comment]
Vietnam is in the period of social distancing because of covid19, so updates and support are limited at this time.

**Comment by mcanevet** [comment]
@banhbaoxamlan I'm wondering where does the OpenCore binaries you are using come from exactly. I'm looking for the commit ref of the OpenCore binary provided in 9b80582 (which works for me) and try to understand what's broken in the commit after that one, but I couldn't find which version you are using. I though it was maybe one the releases here https://github.com/dortania/build-repo, but I compared the checksum of all releases from 2021-05-29 to 2021-06-10, and none matches. Did you build it yourself?

**Comment by mrfiskare** [comment]
> @banhbaoxamlan I'm wondering where does the OpenCore binaries you are using come from exactly. I'm looking for the commit ref of the OpenCore binary provided in [9b80582](https://github.com/banhbaoxamlan/X230-Hackintosh/commit/9b80582f4e471b2f4e3021cc1255444e2e39965b) (which works for me) and try to understand what's broken in the commit after that one, but I couldn't find which version you are using. I though it was maybe one the releases here https://github.com/dortania/build-repo, but I compared the checksum of all releases from 2021-05-29 to 2021-06-10, and none matches. Did you build it yourself?

Hey @mcanevet! I'm revisiting macOS with my X230 (been using the dell 9020 for a while). I'd like to ask you if you ever managed to solve booting Monterey with the latest (or any) build. Thanks in advance!

**Comment by mcanevet** [comment]
@MrFiskare I don't have my x230 anymore, sorry.
